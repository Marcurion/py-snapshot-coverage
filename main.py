class SystemUnderTest:
    def __init__(self, name):
        self.name = name

    def add(self, a: int, b: int):
        return a + b

    def sub(self, a:int, b:int):
        return a - b

    def concat(self, a:str, b:int):
        return f'{a} + {b} = {a + str(b)}'