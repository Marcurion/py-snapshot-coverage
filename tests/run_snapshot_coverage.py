import pytest
import sys
import os
sys.path.append(os.path.dirname(os.path.realpath(__file__)) + "/..")
from generator import Generator
from main import SystemUnderTest

generator = Generator()
system = SystemUnderTest("1")

ints = [1,2,3]
strings = ["abc","def","ghi"]
int_x_int = generator.generate_combinations(None, ints, ints)
ints += [4,5]
int_x_int = generator.generate_combinations(int_x_int, ints, ints)


def test_add_function(snapshot):
        for test_tuple in int_x_int:
            print(test_tuple)
            # 1-n
            assert system.add(*test_tuple) == snapshot

def test_subtract_function(snapshot):
        for test_tuple in int_x_int:
            print(test_tuple)
            assert system.sub(*test_tuple) == snapshot
