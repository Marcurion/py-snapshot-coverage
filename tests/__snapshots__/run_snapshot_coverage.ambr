# serializer version: 1
# name: test_add_function
  2
# ---
# name: test_add_function.1
  3
# ---
# name: test_add_function.10
  6
# ---
# name: test_add_function.11
  6
# ---
# name: test_add_function.12
  7
# ---
# name: test_add_function.13
  7
# ---
# name: test_add_function.14
  8
# ---
# name: test_add_function.15
  5
# ---
# name: test_add_function.16
  6
# ---
# name: test_add_function.17
  7
# ---
# name: test_add_function.18
  8
# ---
# name: test_add_function.19
  9
# ---
# name: test_add_function.2
  4
# ---
# name: test_add_function.20
  6
# ---
# name: test_add_function.21
  7
# ---
# name: test_add_function.22
  8
# ---
# name: test_add_function.23
  9
# ---
# name: test_add_function.24
  10
# ---
# name: test_add_function.3
  3
# ---
# name: test_add_function.4
  4
# ---
# name: test_add_function.5
  5
# ---
# name: test_add_function.6
  4
# ---
# name: test_add_function.7
  5
# ---
# name: test_add_function.8
  6
# ---
# name: test_add_function.9
  5
# ---
# name: test_subtract_function
  0
# ---
# name: test_subtract_function.1
  -1
# ---
# name: test_subtract_function.10
  -4
# ---
# name: test_subtract_function.11
  -2
# ---
# name: test_subtract_function.12
  -3
# ---
# name: test_subtract_function.13
  -1
# ---
# name: test_subtract_function.14
  -2
# ---
# name: test_subtract_function.15
  3
# ---
# name: test_subtract_function.16
  2
# ---
# name: test_subtract_function.17
  1
# ---
# name: test_subtract_function.18
  0
# ---
# name: test_subtract_function.19
  -1
# ---
# name: test_subtract_function.2
  -2
# ---
# name: test_subtract_function.20
  4
# ---
# name: test_subtract_function.21
  3
# ---
# name: test_subtract_function.22
  2
# ---
# name: test_subtract_function.23
  1
# ---
# name: test_subtract_function.24
  0
# ---
# name: test_subtract_function.3
  1
# ---
# name: test_subtract_function.4
  0
# ---
# name: test_subtract_function.5
  -1
# ---
# name: test_subtract_function.6
  2
# ---
# name: test_subtract_function.7
  1
# ---
# name: test_subtract_function.8
  0
# ---
# name: test_subtract_function.9
  -3
# ---
